src/lib/Microsoft.Health.Fhir.Ingest.Template/IotJsonPathContentTemplateFactory.cs

```diff
@@ -26,7 +26,14 @@ public override IContentTemplate Create(TemplateContainer jsonTemplate)
                 throw new InvalidTemplateException($"Expected an object for the template property value for template type {targetTypeName}.");
             }
 
-            return jsonTemplate.Template.ToObject<IotJsonPathContentTemplate>();
+            var iotJsonPathContentTemplate = jsonTemplate.Template.ToObject<IotJsonPathContentTemplate>(GetJsonSerializer());
+            if (SerializationErrors?.Count > 0)

```

Comment:
I did a test.  It looks like the default serializer will stop on the first error so the benefit of this is we get all serialization errors.  I would prefer to modify this so we get a Serialization context that will have a the errors and a reference to the JsonSerializer.  It would also allow you to wrap this error checking and raising logic as part of the context so we have less repeated code.  You also no longer have to bind the errors to the factory now and reset them if there are multiple invocations.  My concern about memory leaks is also gone since we are creating a delegate to link the serializer to a factory with a longer lifetime.